src/server/storage.zig

```diff
@@ -42,41 +46,54 @@ pub const RegionFile = struct { // MARK: RegionFile
 			return self;
 		};
 		defer main.stackAllocator.free(data);
-		if(data.len < headerSize) {
-			std.log.err("Region file {s} is too small", .{path});
-			return self;
-		}
-		var i: usize = 0;
-		const fileVersion = std.mem.readInt(u32, data[i..][0..4], .big);
-		i += 4;
-		const fileSize = std.mem.readInt(u32, data[i..][0..4], .big);
-		i += 4;
+		self.load(data) catch |err| switch(err) {
+			error.CorruptedTooShort => {
+				std.log.err("Corrupted region file: {s} (too short)", .{path});
+			},
+			error.CorruptedTooLong => {
+				std.log.err("Corrupted region file: {s} (too long)", .{path});
+			},
+			error.CorruptedSizeMismatch => {
+				std.log.err("Corrupted region file: {s} (size mismatch)", .{path});
+			},
+			error.VersionNotSupported => {
+				std.log.err("Unsupported version of region file: {s}", .{path});
+			},
```

Comment:
I don't think the error type matters a lot.
What really matters is printing a message for the player, and a stack trace for the developer. The error type itself is mostly meaningless to the user, and the stack trace is most important to the developer.

In my opinion we should just keep user the error from the reader everywhere, to avoid all the ugly `catch return error` statements and potentially make it faster too.